### PR TITLE
feat(storage-browser): add and integrate useGetCredentialsProvider, utils, unit tests

### PR DESCRIPTION
**Pull request by calebpollman**
#### Description of changes
Add and integrate useGetCredentialsProvider, utils, unit tests

#### Issue #, if available
NA

#### Description of how you validated changes
Manual testing, unit tests

#### Checklist

- [x] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [x] PR description included
- [x] `yarn test` passes and tests are updated/added
- [x] PR title and commit messages follow [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) syntax

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: examples/next/pages/ui/components/storage/storage-browser/default-auth/index.page.tsx [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import React from 'react';
 import { Amplify } from 'aws-amplify';
+import { signOut } from 'aws-amplify/auth';
```
[the remark this answers is not in the document]
Comment (jordanvn) [reply examples/next/pages/ui/components/storage/storage-browser/default-auth/index.page.tsx:3]: Why is this import unused here?

[the remark this answers is not in the document]
Comment (calebpollman) [reply examples/next/pages/ui/components/storage/storage-browser/default-auth/index.page.tsx:3]: It should have been integrated with a sign out button, added in last commit

**Review by esauerbo — DISMISSED** [review]

**Review by hbuchel — APPROVED** [review]

File: packages/react-storage/src/components/StorageBrowser/__tests__/createStorageBrowser.spec.tsx [hunk 2]
```diff
@@ -19,4 +20,6 @@ describe('createStorageBrowser', () => {
       expect(render(<StorageBrowser />).container).toBeDefined();
     });
   });
+
+  it.todo('throws when registerAuthListener is not a function');
```
Comment (jordanvn) [inline packages/react-storage/src/components/StorageBrowser/__tests__/createStorageBrowser.spec.tsx:24]: Info: when would this test case be returned to? Should it not be within this set of changes?

File: packages/react-storage/src/components/StorageBrowser/adapters/createAmplifyAuthAdapter.ts [hunk 3]
```diff
@@ -22,7 +22,7 @@ export const createAmplifyAuthAdapter = (input?: {
   const { bucket, region } = Amplify.getConfig()?.Storage?.S3 ?? {};
   if (!bucket || !region) {
     throw new Error(
-      'Amplify Storage configuration not found. Did you run `Amplify.configure` at your project root?'
+      'Amplify Storage configuration not found. Did you run `Amplify.configure` from your project root?'
```
Comment (jordanvn) [inline packages/react-storage/src/components/StorageBrowser/adapters/createAmplifyAuthAdapter.ts:25]: Nice, this looks like it aligns better with existing phrasing 👍

File: turbo.json [hunk 4]
```diff
@@ -1,6 +1,5 @@
 {
   "$schema": "https://turborepo.org/schema.json",
-  "baseBranch": "origin/main",
```
Comment (jordanvn) [inline turbo.json:3]: Nice, this is an issue I was looking into resolving. Glad to see this change occurring

**Review by jordanvn — APPROVED** [review]
These changes all look good to me

**Review by jordanvn — APPROVED** [review]

**Review by hbuchel — APPROVED** [review]

**Pull request merged**